CODINGSTYLE.md

```diff
@@ -0,0 +1,541 @@
+
+Do not fear for over ranges.
+The Scala.js optimizer inlines them away.
+
+
+## if/else
+
+An `if/else` pair in expression position can be written as
+
+```scala
+val x =
+  if (condition) someExpr
+  else anotherExpr
+```
+
+assuming both lines fit in the line length limit.
+We call this formatting a two-liner `if/else`.
+
+Otherwise, both expressions should be put on separate lines, and indented 2 spaces.
+In addition, the rhs of the `=` must then be surrounded in braces:
+
+```scala
+val x = {
+  if (condition)
+    someExpr
+  else
+    anotherExpr
+}
+```
+
+If one of the brances requires braces, then put braces on both branches:
+
+```scala
+val x = {
+  if (condition) {
+    val x = someExpr
+    x + 5
+  } else {
+    anotherExpr
+  }
+}
+```
+
+`if`s and `if/else`s in statement position should always have their branch(es) on dedicated lines:
+
+```scala
+if (index >= size)
+  throw new IndexOutOfBoundsException
+```
+
+If the `condition` of an `if` (or `while`, for that matter) is too long, it can be broken *at most once* with 4 spaces of indentation.
+In that case, the if and else parts must surrounded by braces, even if they are single-line.
+Obviously, the two-liner `if/else` formatting cannot be applied.
+
+If the condition is so long that two lines are not enough, then it should be extracted in a local `def` before it, such as:
+
+```scala
+val ident: String = ???
+
+def isValidIdent = {
+  ident != "" &&
+  ident.charAt(0).isUnicodeIdentifierStart &&
+  ident.tail.forall(_.isUnicodeIdentifierPart)
+}
+
+if (isValidIdent)
+  doSomething()
+else
+  doSomethingElse()
+```
+
+
+## Pattern matching
+
+One-liner cases should be written on one line, and the arrows aligned:
+
+```scala
+x match {
+  case Foo(a, b) => a + b
+  case Bar(y)    => 2 * y
+}
+```
+
+If the body of a case does not fit on the same line, then put the body on the next line, indented 2 spaces, without braces around it.
+In that case, also put blank lines around that `case`, and do not align its arrow with the other groups:
+
+```scala
+x match {
+  case Foo(a, b) => a + b
+
+  case Bar(y) =>
+    if (y < 5) y
+    else y * 2
+}
+```
+
+When pattern-matching based on specific subtypes of a value, reuse the same identifier for the refined binding, e.g.,
+
+```scala
+that match {
+  case that: Foo => that.bar == 5
+  case _         => false
+}
+```
+
+When using type-based pattern matching combined with alternatives (using `|`), remove the space after the `:` operator:
+
+```scala
+that match {
+  case _:Foo | _:Bar | _:Foobar => true
+  case _                        => false
+}
+```
+
+This is an instantiation of the rule saying that spaces can be removed around a binary operator to highlight its higher priority wrt. its neighbors.
+
+As a reminder, avoid pattern-matching on `Option` types.
+Use `fold` instead.
+
+
+## Implementing the Java lib
+
+Special rules apply to implementing classes of the JDK (typically in `javalanglib/` or `javalib/`).
+
```

Comment:
I think we're good on that point. We already have that on the top of these two files:
https://github.com/scala-js/scala-js/blob/master/DEVELOPING.md
https://github.com/scala-js/scala-js/blob/master/CONTRIBUTING.md
Not to mention it has nothing to do with the coding style ^^